### ISSUE TITLE
Sik-Ho Tsang | Brief Review -- Learning Deep Representations by Mutual Information Estimation and Maximization.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#70]
Sik-Ho Tsang. [Brief Review — Learning Deep Representations by Mutual Information Estimation and Maximization](https://sh-tsang.medium.com/brief-review-learning-deep-representations-by-mutual-information-estimation-and-maximization-fe58577d046d).

**Comment by NorbertZheng** [comment]
## Overview
Deep InfoMax (DIM) with Global & Local Objectives.
 - Learning Deep Representations by Mutual Information Estimation and Maximization, Deep InfoMax (DIM), by Microsoft Research 2019 ICLR, Over 1800 Citations. Self-Supervised Learning, Contrastive Learning, Image Classification.

 - Self-supervised representation learning is based on **maximizing mutual information** between features extracted from multiple views of a shared context.
 - While multiple views could be produced by
    - observing it from different locations (e.g., camera positions within a scene),
    - and via different modalities (e.g., tactile, auditory, or visual),
    - an ImageNet image could provide a context from which one produces multiple views by **repeatedly applying data augmentation**.

 - This is a paper from [Prof. Bengio](https://sh-tsang.medium.com/deep-learning-for-ai-by-yoshua-bengio-yann-lecun-and-geoffrey-hinton-23c16a479300) research group.

**Comment by NorbertZheng** [comment]
## Deep InfoMax (DIM)
![image](https://user-images.githubusercontent.com/37235023/221460857-c2870b66-99f5-4bfa-b5a2-7aab3ef6feb8.png)
 - Let $X$ and $Y$ be the domain, e.g. images, and range of a continuous, e.g. feature vector, and (almost everywhere) differentiable parametric function, $E_{\psi}: X\rightarrow Y$ with parameters $\psi$.
 - The encoder should be trained such that the mutual information is maximized:
    - Find the set of parameters, $\psi$, such that the mutual information, $I(X;E_{\psi}(X))$, is maximized.

   Depending on the end-goal, this maximization can be done over the complete input, $X$, or some structured or “local” subset.
 - To maximize MI, a discriminator is trained to classify if the feature vector is real or fake.

**Comment by NorbertZheng** [comment]
### Global DIM: DIM(G)
![image](https://user-images.githubusercontent.com/37235023/221462217-47a265c3-1515-46be-b799-91901c575a02.png)
Deep InfoMax (DIM) with a global $MI(X; Y)$ objective.

 - The above shows Deep InfoMax (DIM) with a global $MI(X, Y)$ objective.
 - Here, Both the high-level feature vector $Y$, and the lower-level $M\times M$ feature map through a discriminator to get the score.
 - Fake samples are drawn by combining the same feature vector with a $M\times M$ feature map from another image.

**Comment by NorbertZheng** [comment]
### Mutual Information Network Estimation
#### Donsker-Varadhan (DV)
One of the approaches follows **Mutual Information Neural Estimation (MINE)** (Belghazi et al., 2018), which uses a lower-bound to the MI based on the Donsker-Varadhan representation (DV, Donsker & Varadhan, 1983) of the KL-divergence:

![image](https://user-images.githubusercontent.com/37235023/221463710-c903edbd-934a-4b1c-bada-9278539bae95.png)

where $T_{\omega}: X\times Y$ is a discriminator function modeled by a neural network with parameters $\omega$.

At a high level, $E$ is optimized by simultaneously estimating and maximizing $I(X;E_{\psi}(X))$:

![image](https://user-images.githubusercontent.com/37235023/221464005-b8371f0c-b55d-41e7-8f30-8300356fae49.png)

where the subscript $G$ denotes “global”.

#### Jensen-Shannon Divergence (JSD)
Jensen-Shannon MI estimator (following the formulation of Nowozin et al., 2016):

![image](https://user-images.githubusercontent.com/37235023/221464287-0809a4f3-5271-46f2-a7e9-162b2c6e9ec0.png)

where $x$ is an input sample, $x$' is an input sampled from $\hat{\mathbb{P}}=\mathbb{P}$, and $sp(z)=log(1+e^{z})$ is the softplus function.

#### Noise-Contrastive Estimation ([NCE](https://sh-tsang.medium.com/review-learning-word-embeddings-efficiently-with-noise-contrastive-estimation-nce-dba1c345c153))
Similar to [NCE](https://sh-tsang.medium.com/review-learning-word-embeddings-efficiently-with-noise-contrastive-estimation-nce-dba1c345c153) or [InfoNCE in CPC](https://sh-tsang.medium.com/review-representation-learning-with-contrastive-predictive-coding-cpc-cpcv1-8ec7c6e91a8e), this loss can also be used with DIM by maximizing:

![image](https://user-images.githubusercontent.com/37235023/221464853-03733e0e-b608-47d5-aabb-03449636eb48.png)

 - It is found that using [InfoNCE](https://sh-tsang.medium.com/review-representation-learning-with-contrastive-predictive-coding-cpc-cpcv1-8ec7c6e91a8e) often outperforms JSD on downstream tasks.

**Comment by NorbertZheng** [comment]
### Local DIM: DIM(L)
![image](https://user-images.githubusercontent.com/37235023/221465541-f6844392-abfb-4c98-a6af-2a0ece688ae0.png)
Maximizing mutual information between local features and global features.

 - First, the image is encoded to a feature map $C_{\psi}(x)$ that reflects some structural aspect of the data, e.g. spatial locality, and this feature map is further summarized into a global feature vector.
 - Then, this feature vector is concatenated with the lower-level feature map at every location.
 - A score is produced for each local-global pair through an additional function.
 - MI estimator is performed on global/local pairs, maximizing the average estimated MI:

![image](https://user-images.githubusercontent.com/37235023/221474519-4f75001e-9e18-402b-a672-03bf009f903e.png)

**Comment by NorbertZheng** [comment]
### Prior Matching
![image](https://user-images.githubusercontent.com/37235023/221474980-4c336bdf-74e1-4081-ac0c-c11cbc19b159.png)
 -  “Real” samples are drawn from a prior while “fake” samples from the encoder output are sent to a discriminator.
 - The discriminator is trained to distinguish between (classify) these sets of samples. The encoder is trained to “fool” the discriminator.
![image](https://user-images.githubusercontent.com/37235023/221475211-b1213cd2-ebc7-4529-91ce-5c9c0c9ab94d.png)
 - With prior matching, encoder can generate features that close to prior.

**Comment by NorbertZheng** [comment]
### Complete Objective
All three objectives — global and local MI maximization and prior matching — can be used together, as the complete objective for Deep InfoMax (DIM):

![image](https://user-images.githubusercontent.com/37235023/221475334-17a7ae07-19cd-4e06-affd-4e37440d1404.png)

**Comment by NorbertZheng** [comment]
## Results
![image](https://user-images.githubusercontent.com/37235023/221475435-c740b280-d8f0-49a6-83a0-7e06d382b5bf.png)
Classification accuracy (top 1) results on CIFAR10 and CIFAR100.

![image](https://user-images.githubusercontent.com/37235023/221475483-64ad83b1-f607-4597-aa5d-929d01b95d9c.png)
Classification accuracy (top 1) results on Tiny ImageNet and STL-10.

In general, DIM with the local objective, DIM(L), outperformed all models presented here by a significant margin on all datasets.

Among DV, JSD & [InfoNCE](https://sh-tsang.medium.com/review-representation-learning-with-contrastive-predictive-coding-cpc-cpcv1-8ec7c6e91a8e), [InfoNCE](https://sh-tsang.medium.com/review-representation-learning-with-contrastive-predictive-coding-cpc-cpcv1-8ec7c6e91a8e) tends to perform best.

![image](https://user-images.githubusercontent.com/37235023/221475616-dae83cba-9d86-49f9-b5aa-6dd441524c5c.png)
Comparisons of DIM with Contrastive Predictive Coding ([CPC](https://sh-tsang.medium.com/review-representation-learning-with-contrastive-predictive-coding-cpc-cpcv1-8ec7c6e91a8e)).

DIM(L) is competitive with [CPC using InfoNCE](https://sh-tsang.medium.com/review-representation-learning-with-contrastive-predictive-coding-cpc-cpcv1-8ec7c6e91a8e).

**Comment by NorbertZheng** [comment]
This is an early paper for self-supervised learning. Many things are tried here.

**Comment by NorbertZheng** [comment]
## Reference
[2019 ICLR] [Deep InfoMax (DIM)]
[Learning Deep Representations by Mutual Information Estimation and Maximization](https://arxiv.org/abs/1808.06670)

**Comment by NorbertZheng** [comment]
## Extended Reading
Unsupervised/Self-Supervised Learning
1993 … 2021 [[MoCo v3](https://sh-tsang.medium.com/review-moco-v3-an-empirical-study-of-training-self-supervised-vision-transformers-4a1a70c53ecd)] [[SimSiam](https://sh-tsang.medium.com/review-simsiam-exploring-simple-siamese-representation-learning-3c84ceb61702)] [[DINO](https://sh-tsang.medium.com/cfddbb4d3549)] [[Exemplar-v1, Exemplar-v2](https://sh-tsang.medium.com/review-what-makes-instance-discrimination-good-for-transfer-learning-120f758b2bf1)] [[Barlow Twins](https://sh-tsang.medium.com/review-barlow-twins-self-supervised-learning-via-redundancy-reduction-967769fafc1)] [[W-MSE](https://sh-tsang.medium.com/review-whitening-for-self-supervised-representation-learning-f4b9ce032c5f)] [[SimSiam+AL](https://sh-tsang.medium.com/458b14fef08f)] [[BYOL+LP](https://sh-tsang.medium.com/review-self-supervised-semi-supervised-learning-for-data-labeling-and-quality-evaluation-dcb60ac07358)] 2022 [[BEiT](https://sh-tsang.medium.com/review-beit-bert-pre-training-of-image-transformers-c14a7ef7e295)] [[BEiT V2](https://sh-tsang.medium.com/review-beit-v2-masked-image-modeling-with-vector-quantized-visual-tokenizers-d7f1d9944995)] [[Masked Autoencoders (MAE)](https://sh-tsang.medium.com/review-masked-autoencoders-are-scalable-vision-learners-b7c42910f7b4)]
